src/m1_facilitator/handlers/DeclaredDepositIntentsHandler.ts

```diff
@@ -0,0 +1,174 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import BigNumber from 'bignumber.js';
+import DepositIntent from '../models/DepositIntent';
+import Message, { MessageStatus, MessageType } from '../models/Message';
+import GatewayRepository from '../repositories/GatewayRepository';
+import DepositIntentRepository from '../repositories/DepositIntentRepository';
+import MessageRepository from '../repositories/MessageRepository';
+import Utils from '../../common/Utils';
+import Logger from '../../m0_facilitator/Logger';
+
+/**
+ * It handles updates from DeclaredDepositIntents entity.
+ */
+export default class DeclaredDepositIntentsHandler {
+  /** Instance of DepositIntentRepository. */
+  private depositIntentRepository: DepositIntentRepository;
+
+  /** Instance of MessageRepository. */
+  private messageRepository: MessageRepository;
+
+  /** Instance of GatewayRepository. */
+  private gatewayRepository: GatewayRepository;
+
+  /**
+   * Constructor for DeclaredDepositIntentHandler.
+   *
+   * @param depositIntentRepository Instance of DepositIntentRepository.
+   * @param gatewayRepository Instance of GatewayRepository.
+   * @param messageRepository Instance of MessageRepository.
+   */
+  public constructor(
+    depositIntentRepository: DepositIntentRepository,
+    gatewayRepository: GatewayRepository,
+    messageRepository: MessageRepository,
+  ) {
+    this.depositIntentRepository = depositIntentRepository;
+    this.gatewayRepository = gatewayRepository;
+    this.messageRepository = messageRepository;
+  }
+
+  /**
+   * It handles DeclaredDepositIntent entity records.
+   * @param records List of DeclaredDepositIntent entity.
+   */
+  public async handle(records: {
+    contractAddress: string;
+    messageHash: string;
+    intentHash: string;
+    tokenAddress: string;
+    beneficiary: string;
+    amount: BigNumber;
+    feeGasPrice: BigNumber;
+    feeGasLimit: BigNumber;
+    blockNumber: BigNumber;
+  }[]): Promise<void> {
+    const promisesCollection = records.map(
+      async (record): Promise<void> => {
+        const isNewMessage = await this.handleMessage(
+          record.contractAddress,
+          record.messageHash,
+          record.feeGasPrice,
+          record.feeGasLimit,
+          record.blockNumber,
+          record.intentHash,
+        );
+        if (isNewMessage) {
+          await this.handleDepositIntent(
+            record.messageHash,
+            record.tokenAddress,
+            record.amount,
+            record.beneficiary,
+            record.intentHash,
+          );
+        }
+      },
+    );
+    await Promise.all(promisesCollection);
+    Logger.debug('Messages saved');
+  }
+
+  /**
+   * It updates Message model.
```

Comment:
```suggestion
   * It creates/updates Message respository.
```

Reply:
Done

Reply:
Reverted this change as per https://github.com/mosaicdao/facilitator/pull/323#discussion_r383291333